File: examples/functions/rest-api-methods/products/[id]/delete.go
package main

import (
	"encoding/json"
	"strconv"
)

// DELETE /products/:id — id merged into event from [id] filename
func handler(event map[string]interface{}) interface{} {
	idStr, _ := event["id"].(string)
	id, _ := strconv.Atoi(idStr)

	body, _ := json.Marshal(map[string]interface{}{
		"id": id, "deleted": true,
	})
	return map[string]interface{}{
		"status":  200,
		"headers": map[string]string{"Content-Type": "application/json"},
		"body":    string(body),
	}
}

File: examples/functions/rest-api-methods/products/[id]/delete.rs
use serde_json::{json, Value};

// DELETE /products/:id — id merged into event from [id] filename
pub fn handler(event: Value) -> Value {
    let id: i64 = event["id"].as_str().unwrap_or("0").parse().unwrap_or(0);

    json!({
        "status": 200,
        "headers": { "Content-Type": "application/json" },
        "body": serde_json::to_string(&json!({
            "id": id, "deleted": true
        })).unwrap()
    })
}
